### ISSUE TITLE
homography_warp: normalized_homography=False silently ignores mode and align_corners

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4772]
## Describe the bug

`kornia.geometry.transform.homography_warp(..., normalized_homography=False)` accepts `mode` and `align_corners` but silently ignores both. The pixel-homography branch delegates to `warp_perspective` with both values hardcoded (`kornia/geometry/transform/imgwarp.py`, end of `homography_warp`):

```python
return warp_perspective(
    patch_src, src_homo_dst, dsize, mode="bilinear", padding_mode=padding_mode, align_corners=True
)
```

`padding_mode` is forwarded; `mode` and `align_corners` are not. The function's own default is `align_corners=False`, so a plain call with `normalized_homography=False` also gets a different convention than the signature advertises.

## Reproduction

```python
import torch
from kornia.geometry.transform import homography_warp, warp_perspective

img = torch.arange(16.0).view(1, 1, 4, 4)
H = torch.tensor([[[1.0, 0.0, 1.5], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]])  # src->dst pixels, shift right 1.5 px

# align_corners is ignored: visible where the two conventions differ, i.e. out-of-bounds handling
hw = homography_warp(img, H, (4, 4), padding_mode="reflection", align_corners=False, normalized_homography=False)
wp_false = warp_perspective(img, H, (4, 4), padding_mode="reflection", align_corners=False)
wp_true = warp_perspective(img, H, (4, 4), padding_mode="reflection", align_corners=True)
print((hw - wp_false).abs().max())  # 1.0  -- should be ~0
print((hw - wp_true).abs().max())   # 0.0  -- it silently used align_corners=True

# mode is ignored
hn = homography_warp(img, H, (4, 4), mode="nearest", normalized_homography=False)
print(hn[0, 0, 0])                                            # [0.0, 0.0, 0.5, 1.5] -- bilinear values
print(warp_perspective(img, H, (4, 4), mode="nearest")[0, 0, 0])  # [0.0, 0.0, 0.0, 2.0]
```

## Expected behavior

With `normalized_homography=False`, `homography_warp` documents that it consumes the source-to-destination pixel homography "exactly like `warp_perspective`". It should then forward `mode` and `align_corners` to `warp_perspective` as it already forwards `padding_mode`, so the call above equals `warp_perspective(img, H, (4, 4), mode=mode, padding_mode=padding_mode, align_corners=align_corners)`. Alternatively, if the forced values are intended, the function should reject or warn on a non-default `mode`/`align_corners` in that branch rather than drop them silently. Nothing in the history (#1438, which added the branch) states that the hardcoding is deliberate.

Note that forwarding `align_corners` changes the default-argument path of `normalized_homography=False` calls (from `True` to the signature's `False`), but only in out-of-bounds handling (the 2-D normalization is consistent since #3945, so in-bounds samples agree to roundoff).

## Environment

kornia `main` at 57cd68e6f, torch 2.14.0, CPU.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
